compiler/modules/front.syntax/parseExpr.bal

```diff
@@ -99,18 +99,33 @@ function parseRelationalExpr(Tokenizer tok) returns Expr|err:Syntax {
         return bin;
     }
     else if t == "is" {
-        tok.setMode(MODE_TYPE_DESC);
-        check tok.advance();
-        InlineTypeDesc td = check parseInlineTypeDesc(tok);
-        tok.setMode(MODE_NORMAL);
+        InlineTypeDesc td = check handleTypeDesc(tok);
         TypeTestExpr typeTest = { td, left: expr, semType: resolveInlineTypeDesc(td) };
         return typeTest;
     }
+    if t == "!" {
+        check tok.advance();
+        Token? t2 = tok.current();
+        if t2 is "is" {
+            InlineTypeDesc notTd = check handleTypeDesc(tok);
+            TypeTestNotExpr typeTest = { notTd, left: expr, semType: resolveInlineTypeDesc(notTd) };
+            return typeTest;
+        }
+        return err:syntax("invalid operator");
+    }
     else {
         return expr;
     }
 }
 
+function handleTypeDesc(Tokenizer tok) returns InlineTypeDesc|err:Syntax {
```

Comment:
Please choose a name consistent with other names in this file.

Reply:
fixed